stl/inc/chrono

```diff
@@ -2209,6 +2221,2949 @@ namespace chrono {
+                return false;
+            }
+
+            if (_Day_of_year <= 31) {
+                return _Update(_Day, _Yday) && _Update(_Month, 1);
+            }
+
+            const int _Feb_end{_Is_leap(_Year) ? 60 : 59};
+            if (_Day_of_year <= _Feb_end) {
+                return _Update(_Day, _Yday - 31) && _Update(_Month, 2);
+            }
+
+            // Shift day-of-year so that 1 == March 1. This is the same as year_month_day::_Civil_from_days, except
+            // _Day_of_year --> _Shifted_yday-1 and _Mp --> _Month - 3.
+            const int _Shifted_yday{*_Day_of_year - _Feb_end};
+            return _Update(_Month, (535 * _Shifted_yday + 48950) >> 14)
+                && _Update(_Day, _Shifted_yday - ((979 * *_Month - 2918) >> 5));
+        }
+
+        static constexpr int _Era_begin_wday{3}; // Wednesday
+
+        _NODISCARD static constexpr int _Jan1_weekday(int _Year) {
+            --_Year;
+            const int _Era = (_Year >= 0 ? _Year : _Year - 399) / 400;
+            const int _Yoe = _Year - _Era * 400;
+            // Jan. 1 is always day 306 of the shifted [Mar, ..., Dec, Jan, Feb] year.
+            const int _Doe = ((1461 * _Yoe) >> 2) - _Yoe / 100 + 306;
+            return (_Doe + _Era_begin_wday) % 7;
+        }
+
+        _NODISCARD static constexpr int _Iso8601_weeks(int _Year) {
+            const int _P_y = (_Year + _Year / 4 - _Year / 100 + _Year / 400) % 7;
+            --_Year;
+            const int _P_ym1 = (_Year + _Year / 4 - _Year / 100 + _Year / 400) % 7;
+            return 52 + (_P_y == 4 || _P_ym1 == 3);
+        }
+
+        _NODISCARD static constexpr int _Iso8601_week(const int _Day_of_year, const int _Weekday, const int _Year) {
+            // Jan. 4 is always week 1; rollover to next week always happens on Monday.
+            const auto _Week{(_Day_of_year + 9 - _Prev_weekday(_Weekday, 1)) / 7};
+            if (_Week < 1) {
+                return _Iso8601_weeks(_Year - 1);
+            } else if (_Week > _Iso8601_weeks(_Year)) {
+                return 1;
+            } else {
+                return _Week;
+            }
+        }
+
+        _NODISCARD static constexpr bool _Is_leap(const int _Year) {
+            return _Year % 4 == 0 && (_Year % 100 != 0 || _Year % 400 == 0);
+        }
+
+        _NODISCARD static constexpr int _Days_in_year(const int _Year) {
+            return _Is_leap(_Year) ? 366 : 365;
+        }
+
+        _NODISCARD static constexpr int _Next_weekday(const int _Wday, const int _Shift) {
+            // 0 <= _Shift <= 6
+            int _Result = _Wday + _Shift;
+            if (_Result >= 7) {
+                _Result -= 7;
+            }
+            return _Result;
+        }
+
+        _NODISCARD static constexpr int _Prev_weekday(const int _Wday, const int _Shift) {
+            // 0 <= _Shift <= 6
+            return (_Wday >= _Shift ? 0 : 7) + (_Wday - _Shift);

```

Comment:
I really dislike, that we write the essentially same algorithm totally different here

Reply:
We can consider PRs in the future to centralize some of this code, if doing so ends up reducing complexity. For now I think we should proceed with the code as-is - there are no ABI concerns so we can refactor later.